modin/pandas/groupby.py

```diff
@@ -924,6 +979,19 @@ def _apply_agg_function(self, f, *args, **kwargs):
             f, dict
         ), "'{0}' object is not callable and not a dict".format(type(f))
 
+        internal_by = self._get_internal_by()
```

Comment:
this is a full code duplication from `_wrap_aggregation` because these functions are actually equal. In #3197 `_apply_agg_function` will be removed as well as this duplication